src/Type/ArrayType.php

```diff
@@ -432,55 +429,15 @@ public function toArray(): Type
 		return $this;
 	}
 
-	/** @deprecated Use getArraySize() instead */
-	public function count(): Type
+	public function toArrayKey(): Type
 	{
-		return $this->getArraySize();
+		return new ErrorType();
 	}
 
-	public static function castToArrayKeyType(Type $offsetType): Type
```

Comment:
Removing the method is a BC break, we should keep it but simply call `toArrayKey` on the type. And mark it as deprecated.